src/webui/api/torrentscontroller.cpp

```diff
@@ -2072,16 +2072,14 @@ void TorrentsController::parseMetadataAction()
     if (uploadedTorrents.isEmpty())
         throw APIError(APIErrorType::BadParams, tr("Must specify torrent file(s)"));
 
-    QJsonObject result;
+    QJsonArray result;
     for (auto it = uploadedTorrents.constBegin(); it != uploadedTorrents.constEnd(); ++it)
     {
         if (const auto loadResult = BitTorrent::TorrentDescriptor::load(it.value()))
         {
             const BitTorrent::TorrentDescriptor &torrentDescr = loadResult.value();
             m_torrentMetadataCache.insert(torrentDescr.infoHash().toTorrentID(), torrentDescr);
-
-            const QString &fileName = it.key();
-            result.insert(fileName, serializeTorrentInfo(torrentDescr));
+            result.append(serializeTorrentInfo(torrentDescr));
```

Comment:
Isn't it expected that it returns array of the same size as input one? How would the client be able to match the received data with the sent files?

Reply:
We can still respond with an object if we wanted to, since the WebUI now uses an integer instead of the actual file name. The upside is that then this PR has no breaking changes, but the downside is other clients using the WebAPI will have to be aware of the utf8 limitations.

Reply:
> > It always return an array of the same size or throws an error.
> 
> Oh, damn it, it's just that I never approved of this behavior, so I thought it would work differently.

This refers to the fact that entire batch fails due to a single problematic file. I have always believed that similar (batch) endpoints should handle as many jobs as possible and return statuses of failed ones as part of result.


> the downside is other clients using the WebAPI will have to be aware of the utf8 limitations.

I think this is not the only possible drawback of relying on provided filenames since their behavior isn't strictly defined.

Reply:
> I have always believed that similar (batch) endpoints should handle as many jobs as possible and return statuses of failed ones as part of result.

This is the existing behavior so we're not going to be changing it in this PR.

> I think this is not the only possible drawback of relying on provided filenames since their behavior isn't strictly defined.

I really don't understand what you're asking. Can you be more clear? What remaining changes do you need before this PR is approved?